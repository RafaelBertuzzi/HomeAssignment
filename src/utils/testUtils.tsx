import { ReactNode } from 'react';
import renderer from 'react-test-renderer';

import { LayoutProvider } from '@components/LayoutProvider';
import { render } from '@testing-library/react-native';

export const rendererWithProviders = (children: ReactNode) => {
  return renderer.create(<LayoutProvider>{children}</LayoutProvider>).toJSON();
};

export const renderWithProviders = (children: ReactNode) => {
  return render(<LayoutProvider>{children}</LayoutProvider>).toJSON();
};
