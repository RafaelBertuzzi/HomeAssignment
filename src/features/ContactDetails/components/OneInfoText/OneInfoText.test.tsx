import { cleanup } from '@testing-library/react-native';
import { rendererWithProviders } from '@utils/testUtils';

import { OneInfoText } from '.';

describe('Contact Details Screen Component: OneInfoText', () => {
  afterEach(cleanup);

  it('should render', () => {
    const json = rendererWithProviders(
      <OneInfoText label="Test" value={'1'} />,
    );
    expect(json).toMatchSnapshot();
  });
});
