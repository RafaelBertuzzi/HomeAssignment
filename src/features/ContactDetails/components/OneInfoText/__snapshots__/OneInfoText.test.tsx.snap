// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact Details Screen Component: OneInfoText should render 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    dataSet={{}}
    style={
      {
        "flexDirection": "row",
      }
    }
  >
    <Text
      dataSet={{}}
      style={
        {
          "backgroundColor": undefined,
          "color": "#171717",
          "fontFamily": undefined,
          "fontSize": 18,
          "fontStyle": "normal",
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 27,
          "textDecorationLine": undefined,
        }
      }
    >
      Test:
    </Text>
    <View
      dataSet={{}}
      style={
        {
          "width": 16,
        }
      }
    />
    <Text
      dataSet={{}}
      style={
        {
          "backgroundColor": undefined,
          "color": "#171717",
          "fontFamily": undefined,
          "fontSize": 18,
          "fontStyle": "normal",
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 27,
          "textDecorationLine": undefined,
        }
      }
    >
      1
    </Text>
  </View>
</RNCSafeAreaProvider>
`;
